### PR TITLE
Pix mesh shader output instrumentation

### PR DESCRIPTION
**Pull request by jeffnn**
This pass adds instrumentation similar to the PIX debugging pass that outputs mesh shader output to a  UAV for subsequent retrieval and display by PIX

### CODE REVIEW

File: lib/DxilPIXPasses/DxilPIXMeshShaderOutputInstrumentation.cpp [hunk 1]
```diff
@@ -0,0 +1,374 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxilPIXMeshShaderOutputInstrumentation.cpp // Copyright (C) Microsoft
+// Corporation. All rights reserved.                 // This file is distributed
```
Comment (jpporto) [inline lib/DxilPIXPasses/DxilPIXMeshShaderOutputInstrumentation.cpp]: what is happening here? I am seeing the comment poorly formatted. Is it just the browser failing to render?

File: lib/DxilPIXPasses/DxilPIXMeshShaderOutputInstrumentation.cpp [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+    auto &FunctionUse = *FI++;
+    auto FunctionUser = FunctionUse.getUser();
+
+    auto Call = cast<CallInst>(FunctionUser);
+
+    IRBuilder<> Builder2(Call);
+    BuilderContext BC2{M, DM, Ctx, HlslOP, Builder2};
+
+    {
+      constexpr uint32_t DwordCount = 1   // index indicator
```
Comment (jpporto) [inline lib/DxilPIXPasses/DxilPIXMeshShaderOutputInstrumentation.cpp]: As a suggestion, you could have a helper method, e.g.,

template <typename... T>
void Instrument(BuilderContext &BC, T... values)
{
    llvm::SmallVector<llvm::Value *, 10> Values({static_cast<llvm::Value *>(T)...});
    const uint32_t DwordCount = Values.size();
    llvm::Value * byteOffset = reserveDebugEntrySpace(BC, DwordCount * sizeof(uint323_t);
    for (llvm::Value *V : Values)
    {
        byteOffset = writeDwordAndReturnNewOffset(BC, byteOffset, V);
    }
}

File: lib/DxilPIXPasses/DxilPIXMeshShaderOutputInstrumentation.cpp (hunk 1 above)
Comment (jeffnn) [reply lib/DxilPIXPasses/DxilPIXMeshShaderOutputInstrumentation.cpp]: ! 
I don't know what happened. I re-formatted it.

File: lib/DxilPIXPasses/DxilPIXMeshShaderOutputInstrumentation.cpp (hunk 2 above)
Comment (jeffnn) [reply lib/DxilPIXPasses/DxilPIXMeshShaderOutputInstrumentation.cpp]: That's quite elegant. I'll do so. Thanks.

**Pull request merged**